node.wake

```diff
@@ -27,8 +27,8 @@ global def addNodeEnv packageDir plan =
 
     # Add the installed modules to PATH and NODE_PATH
     plan
-    | addPlanEnvironmentPath  "PATH" "build/{packageDir}/node_modules/.bin"
-    | addPlanEnvironmentPath  "NODE_PATH"  "build/{packageDir}/node_modules"
+    | addPlanRelativePath  "PATH" "build/{packageDir}/node_modules/.bin"
```

Comment:
I'm not sure this is a good idea. This will break if you change the plan's working directory after calling this function. E.g.

```wake
# This
plan
| setPlanDirectory foo
| addPlanRelativePath "PATH" bar

# is not equal to this:
plan
| addPlanRelativePath "PATH" bar
| setPlanDirectory foo
```

This because less obvious if you return a plan from one function and modify it in another.

@jackkoenig, how are you handling this for the analogous case of Scala class paths? I believe the problem with absolute paths is that Wake no longer tracks the files when they are absolute paths, even if they are within the Wake workspace?

Reply:
In api-scala-sifive, paths are relative to the workspace root.

Reply:
To clarify, all paths being created by these functions are relative. By default, they are relative to the workspace. There are no absolute paths.

This repository deals only with Python, Ruby and Node applications. 

At issue are all file paths passed in environment variables or embedded in command line arguments. If we call setPlanDirectory, then those file paths are invalid. This change corrests the strings assigned to various PATH variables during the "addXxxxxEnv" functions. It does not update other PATH variables nor does it change the command line arguments. Those steps remain the responsibility of the caller.

As a more complete solution, we could try using delayed evaluation of  strings which embed file paths. Then all strings, command line or environmental, could be updated automatically.

Scala potentially has similar issues.  My impression is, we do not currently use setPlanDirectory with Scala applications, so having incorrect paths in CLASSPATH  hasn't become a problem yet.

With respect to Scribble, we will soon remove the need to set working directories. We could view this PR as a temporary fix to a temporary problem. Or, we can view it as the first step in solving the more complete problem.